vault/resource_aws_secret_backend_static_role.go

```diff
@@ -94,12 +96,21 @@ func readAWSStaticRoleResource(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 
-	rolePath := d.Id()
-	log.Printf("[DEBUG] Reading %q", rolePath)
+	path := d.Id()
+	log.Printf("[DEBUG] Reading %q", path)
+	pathPieces := strings.Split(path, "/")
+	if len(pathPieces) < 3 || pathPieces[len(pathPieces)-2] != "static-roles" {
```

Comment:
I think this will fail for mount paths with arbitrary slashes, e.g. `foo/bar/static-roles/baz` where `foo/bar` is the mount path.

We could use a regex like https://github.com/hashicorp/terraform-provider-vault/blob/5e8979d3b444a1aa44dcc1af0f15213f6e48997e/vault/resource_kubernetes_secret_backend_role.go#L241

Reply:
+1 one using regexes to get this info! We can never assume what kind of paths the customers will be using and I have made this mistake in the past before 😅

Reply:
The role name can also contain slashes

Reply:
a quick question, what should "a/static-roles/static-roles/b" parse to?

Reply:
Given a regex like `regexp.MustCompile("^(.+)/static-roles/.+$")`, `a/static-roles/static-roles/b` would parse to:

mount name: `a/static-roles`
role name: `b`

I think that is sufficient. We can't stop a user from completely shooting themselves in the foot.

Reply:
I added a regex (`(.+)/static-roles/(.+)`) 

One could probably argue for a narrowing of the dot or adding $/^ marks.

Reply:
Are there any edge cases we would hit if we don't add the beginning and end of line anchors?